village/src/components/SmurfForm.js

```diff
@@ -10,21 +10,25 @@ class SmurfForm extends Component {
     };
   }
 
-  addSmurf = event => {
-    event.preventDefault();
-    // add code to create the smurf using the api
-
-    this.setState({
-      name: '',
-      age: '',
-      height: ''
-    });
-  }
 
   handleInputChange = e => {
     this.setState({ [e.target.name]: e.target.value });
   };
+  addSmurf = () => {
+    const name = {
+      name: this.state.name,
+      age: this.state.age,
+      height: this.state.height
+    };
 
+    axios
+      .post('http://localhost:3333/smurfs', name)
+      .then(response => {
+        this.setState({smurfs: response.data });
```

Comment:
Here's another example of where object destructuring can really come in handy:
```javascript
.then(({ data }) => {
  this.setState({ smurfs: data });
})
```
or even this, changing the variable name in order to use some [ES6 shorthand syntax](https://ariya.io/2013/02/es6-and-object-literal-property-value-shorthand), which you saw earlier in my first destructuring example:
```javascript
.then(({ smurfs }) => {
  this.setState({ smurfs });
})
```